source/partial-on-undefined-deep.d.ts

```diff
@@ -0,0 +1,51 @@
+import type {BuiltIns} from './internal';
+import type {Merge} from './merge';
+
+/**
+Create a deep version of another type where all keys accepting `undefined` type are set to optional.
+
+This utility type is recursive, transforming at any level deep, and recurses into arrays too.
```

Comment:
Is it really expected/wanted to recurse into arrays by default? Maybe we should make it opt-in instead? Like a `recurseIntoArrays` option. Similar to https://github.com/sindresorhus/type-fest/commit/37774694ba8c9ce83a998b317ee560bcff08eb2e but should be `false` by default.

Reply:
I see, both solutions seems fine for me, I will update the type to be consistent with `PartialDeep` asap then :blush: